File: 프로그래머스/1/12928. 약수의 합/약수의 합.py
import math

def solution(n):
    root = int(math.sqrt(n))
    result = 0
    
    for i in range(1, root + 1):
        if (n % i == 0):
            result += i
            if (n / i != i):
                result += int(n / i)
    
    return result

File: 프로그래머스/1/12928. 약수의 합/약수의 합.java
class Solution {
    public int solution(int n) {
        int root = (int)Math.sqrt(n);
        int result = 0;
    
        for (int i = 1; i <= root; i++)
        if (n % i == 0) {
            result += i;
            if (n / i != i)
                result += (int)n / i;
            }
    
    return result;
    }
}